lib/fastlane/plugin/wpmreleasetoolkit/actions/common/check_for_toolkit_updates_action.rb

```diff
@@ -0,0 +1,77 @@
+require 'fastlane/action'
+require 'rubygems/command_manager'
+
+module Fastlane
+  module Actions
+    class CheckForToolkitUpdatesAction < Action
+      TOOLKIT_SPEC_NAME = 'fastlane-plugin-wpmreleasetoolkit'.freeze
+
+      def self.run(params)
```

Comment:
Would we also want a `no_update` option similar to [fastlane's own update](https://github.com/fastlane/fastlane/blob/master/fastlane/lib/fastlane/actions/update_fastlane.rb#L11)? Even if we wouldn't want to pass this in most cases, maybe it'll help when/if there is a bug in this updater and we don't want to change the release toolkit during a code freeze or release, what do you think?

Reply:
So this action does 2 things:
 - Check if there's an update available, and shows an informative message of the outcome (new version found or not)
 - Suggest to update, prompt us if we want to, and if we accept, run `bundle update`

The second step is gated by the `skip_update_suggestion` parameter.
If we want to add a param to gate the first step (and thus also the second step since it doesn't make sense to do the second step without the first, it's either 1+2, or just 1, or nothing), it's easier to just not call the action at all.

So for these cases, I suggest to use something like that in the client app's Fastfile:
```ruby
check_for_toolkit_updates(…) unless ENV['SKIP_TOOLKIT_UPDATES']
```

Which is actually how Fastlane does it via their `FASTLANE_SKIP_UPDATE_CHECK` env var, like [here](https://github.com/fastlane/fastlane/blob/master/fastlane_core/lib/fastlane_core/update_checker/update_checker.rb#L14)

Reply:
As for the `params[:no_update]` that Fastlane's own code is using, it's a little bit different. If you look at the documentation for that param and how it's used in their action, this param is not intended to be used by the users directly, but instead is used internally so that when Fastlane auto-restart itself after a successful update – by calling `exec(…)` [here](https://github.com/fastlane/fastlane/blob/master/fastlane/lib/fastlane/actions/update_fastlane.rb#L82) to spawn the same process again, they don't redo a check on the second run that gets automatically triggered there.

In my action I chose not to go the `exec` route to auto-restart `fastlane` after the update, because I preferred to let us check the changes to the `Gemfile.lock` – and also give the possibility to make additional changes to our `Fastfile` in case we wanted to bump to a next major version with breaking changes – before restarting manually.

Reply:
That all makes a lot of sense, thanks a lot for the detailed explanation!

I am sorry for my lack of knowledge & experience on Ruby/fastlane front and I hope you don't mind these questions (or possibly incorrect) suggestions.

Reply:
I don't mind at all, and please keep asking questions, that's how we learn!